com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Math/Derivative/DDXNode.cs

```diff
@@ -7,20 +7,46 @@ internal class DDXNode : IStandardNode
         public static string Name = "DDX";
         public static int Version = 1;
 
-        public static FunctionDescriptor FunctionDescriptor => new(
+        public static NodeDescriptor NodeDescriptor => new(
             Version,
             Name,
-            "Out = ddx(In);",
-            new ParameterDescriptor("In", TYPE.Vector, Usage.In),
-            new ParameterDescriptor("Out", TYPE.Vector, Usage.Out)
+            new FunctionDescriptor[] {
+                new (
+                    1,
+                    "Standard",
+                    "Out = ddx(In);",
+                    new ParameterDescriptor("In", TYPE.Vector, Usage.In),
```

Comment:
By default this is a float/Vector1 input in old SG, is it okay that we're changing it to a Vector4 here by default? 
When we have casting support between types it probably won't matter, Im just thinking about when we want to upgrade old SG graphs it might be a bit weird if the port type has changed

Reply:
In the sg1 code, it's a `DynamicDimensionVector ` type for the Input and output port, and the UI level it defaults to a float input,  I guess it's just like the _Multiply node_ in the old code, it should be able to take in any kind of vectors but the default is set to float. I thought we are using `Type.Vecto`r for the ports that we want then to be dynamic, I can change it to a Vec4 if that's not the case.

Reply:
This is done correctly.  Currently, parameters of type Vector all default to Vec4.  Once the dynamic vector system is working as it should, they'll default to float - so it will work exactly like the previous version once that system is in place.

Reply:
Ahh I see that makes sense, thanks!!